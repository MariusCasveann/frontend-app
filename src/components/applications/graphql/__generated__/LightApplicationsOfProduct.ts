/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: LightApplicationsOfProduct
// ====================================================

export interface LightApplicationsOfProduct_lightApplicationsOfProduct {
    __typename: 'LightApplicationDTO';
    id: any | null;
    name: string | null;
    service: boolean;
    staticPictureId: any | null;
}

export interface LightApplicationsOfProduct {
    lightApplicationsOfProduct: (LightApplicationsOfProduct_lightApplicationsOfProduct | null)[] | null;
}

export interface LightApplicationsOfProductVariables {
    productId?: any | null;
}
