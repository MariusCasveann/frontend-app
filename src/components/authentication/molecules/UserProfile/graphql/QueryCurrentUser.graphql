query QueryCurrentUser {
    currentUser {
        activated
        createdBy
        createdDate
        email
        firstName
        id
        imageContent
        langKey
        lastModifiedBy
        lastModifiedDate
        lastName
        login
        roles {
            id
            name
            readOnly
            concretePermissions {
                permissionType
                id
                entityType
                entityId
            }
        }
        staticPictureId
        subscribedToNewsletter
    }
}
