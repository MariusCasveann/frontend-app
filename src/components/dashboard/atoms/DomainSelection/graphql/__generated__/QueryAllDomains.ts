/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: QueryAllDomains
// ====================================================

export interface QueryAllDomains_allDomains {
    __typename: 'Domain';
    id: any | null;
    name: string | null;
}

export interface QueryAllDomains {
    allDomains: (QueryAllDomains_allDomains | null)[] | null;
}
