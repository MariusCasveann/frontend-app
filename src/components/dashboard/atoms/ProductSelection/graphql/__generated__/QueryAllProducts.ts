/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: QueryAllProducts
// ====================================================

export interface QueryAllProducts_allProducts {
    __typename: 'LightProductDTO';
    id: any | null;
    logo: string | null;
    name: string | null;
}

export interface QueryAllProducts {
    allProducts: (QueryAllProducts_allProducts | null)[] | null;
}
