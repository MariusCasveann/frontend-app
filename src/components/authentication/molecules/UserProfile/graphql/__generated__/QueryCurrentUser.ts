/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { PermissionTypeEnum, SecuredEntityTypeEnum } from './../../../../../../model/__generated__/globalTypes';

// ====================================================
// GraphQL query operation: QueryCurrentUser
// ====================================================

export interface QueryCurrentUser_currentUser_roles_concretePermissions {
    __typename: 'EntityPermission';
    permissionType: PermissionTypeEnum | null;
    id: any | null;
    entityType: SecuredEntityTypeEnum | null;
    entityId: any | null;
}

export interface QueryCurrentUser_currentUser_roles {
    __typename: 'Role';
    id: any | null;
    name: string | null;
    readOnly: boolean | null;
    concretePermissions: (QueryCurrentUser_currentUser_roles_concretePermissions | null)[] | null;
}

export interface QueryCurrentUser_currentUser {
    __typename: 'UserDTO';
    activated: boolean;
    createdBy: string | null;
    createdDate: any | null;
    email: string | null;
    firstName: string | null;
    id: any | null;
    imageContent: string | null;
    langKey: string | null;
    lastModifiedBy: string | null;
    lastModifiedDate: any | null;
    lastName: string | null;
    login: string | null;
    roles: (QueryCurrentUser_currentUser_roles | null)[] | null;
    staticPictureId: any | null;
    subscribedToNewsletter: boolean | null;
}

export interface QueryCurrentUser {
    currentUser: QueryCurrentUser_currentUser | null;
}
