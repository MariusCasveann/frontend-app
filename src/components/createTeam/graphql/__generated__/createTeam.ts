/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { TeamInput } from './../../../../model/__generated__/globalTypes';

// ====================================================
// GraphQL mutation operation: CreateTeam
// ====================================================

export interface CreateTeam_createTeam {
    __typename: 'TeamDTO';
    id: any | null;
}

export interface CreateTeam {
    createTeam: CreateTeam_createTeam | null;
}

export interface CreateTeamVariables {
    productId?: any | null;
    team?: TeamInput | null;
}
