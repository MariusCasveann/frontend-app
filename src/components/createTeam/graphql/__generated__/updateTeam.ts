/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { TeamInput } from './../../../../model/__generated__/globalTypes';

// ====================================================
// GraphQL mutation operation: UpdateTeam
// ====================================================

export interface UpdateTeam_updateTeam {
    __typename: 'TeamDTO';
    id: any | null;
}

export interface UpdateTeam {
    updateTeam: UpdateTeam_updateTeam | null;
}

export interface UpdateTeamVariables {
    productId?: any | null;
    team?: TeamInput | null;
}
