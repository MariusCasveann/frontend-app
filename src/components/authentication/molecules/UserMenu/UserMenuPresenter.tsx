import * as React from 'react';
import { Flyout, Icon } from 'backoffice-shared-components';
import { LogoutButtonPresenter } from './../../../authentication';
import { Button } from 'antd';
import './UserMenu.css';
import { createHashHistory } from 'history';
import { QueryCurrentUser_currentUser } from '../UserProfile/graphql/__generated__/QueryCurrentUser';

export const history = createHashHistory();

interface UserMenuPresenter {
    currentUser: QueryCurrentUser_currentUser;
}

const redirectToProfilePage = () => {
    history.push(`/user-edit`);
};

export default ({ currentUser }: UserMenuPresenter) => {
    if (currentUser) {
        return (
            <div className="user-menu">
                <Flyout
                    content={<span>{currentUser.email}</span>}
                    footer={
                        <span>
                            <Button type="link" onClick={redirectToProfilePage}>
                                Profile
                            </Button>
                            <br />
                            <Icon type="power-off" size="small" color="black" className="margins-power-icon" />
                            <LogoutButtonPresenter />
                        </span>
                    }
                    label={
                        <span>
                            {currentUser.firstName} {currentUser.lastName}
                        </span>
                    }
                    position="left"
                />
            </div>
        );
    }
    return null;
};
