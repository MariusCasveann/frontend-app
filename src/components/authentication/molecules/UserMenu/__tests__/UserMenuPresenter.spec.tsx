import * as React from 'react';
import renderer from 'react-test-renderer';
jest.mock('antd', () => ({ Button: 'Button' }));
jest.mock('backoffice-shared-components', () => ({ Flyout: 'Flyout', Icon: 'Icon' }));
jest.mock('./../../../../authentication', () => ({ LogoutButtonPresenter: 'LogoutButtonPresenter' }));
import UserMenuPresenter from '../UserMenuPresenter';
import { QueryCurrentUser_currentUser } from '../../UserProfile/graphql/__generated__/QueryCurrentUser';

const mockData: QueryCurrentUser_currentUser = {
    __typename: 'UserDTO',
    activated: true,
    createdBy: 'anonymousUser',
    createdDate: '2019-05-27T10:53:29.944312Z',
    email: 'test@mediamarktsaturn.com',
    firstName: 'Test',
    id: 49050,
    imageContent: null,
    langKey: null,
    lastModifiedBy: 'Test',
    lastModifiedDate: '2019-06-28T05:56:51.175813Z',
    lastName: 'Test',
    login: 'Test',
    roles: null,
    staticPictureId: null,
    subscribedToNewsletter: true
};

describe('UserMenuPresenter', () => {
    it.only('renders correctly', () => {
        const tree = renderer.create(<UserMenuPresenter currentUser={mockData} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
