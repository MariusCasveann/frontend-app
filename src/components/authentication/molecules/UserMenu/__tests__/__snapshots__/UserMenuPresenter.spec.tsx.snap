// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMenuPresenter renders correctly 1`] = `
<div
  className="user-menu"
>
  <Flyout
    content={
      <span>
        test@mediamarktsaturn.com
      </span>
    }
    footer={
      <span>
        <Button
          onClick={[Function]}
          type="link"
        >
          Profile
        </Button>
        <br />
        <Icon
          className="margins-power-icon"
          color="black"
          size="small"
          type="power-off"
        />
        <LogoutButtonPresenter />
      </span>
    }
    label={
      <span>
        Test
         
        Test
      </span>
    }
    position="left"
  />
</div>
`;
